internal/printer/printer_test.go

```diff
@@ -1665,6 +1665,13 @@ import { Container, Col, Row } from 'react-bootstrap';
 				code: `${$$maybeRenderHead($$result)}<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"><rect xlink:href="#id"></rect></svg>`,
 			},
 		},
+		{
+			name:   "Signal when using namespaced attributes with expression values",
+			source: `<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"><rect xlink:href={` + BACKTICK + `#${iconId}` + BACKTICK + `}></svg>`,
+			want: want{
+				code: `${$$maybeRenderHead($$result)}<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"><rect xlink:${$$addAttribute(` + BACKTICK + `#${iconId}` + BACKTICK + `, "href", undefined, false)}></rect></svg>`,
```

Comment:
```suggestion
				code: `${$$maybeRenderHead($$result)}<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"><rect ${$$addAttribute(` + BACKTICK + `#${iconId}` + BACKTICK + `, "xlink:href")}></rect></svg>`,
```

Shouldn't the output be ideally something like this? Where the key is `xlink:href`, instead of `href`.

Reply:
Ideally yes, and that would avoid making a change to the `addAttribute` in the runtime.
But currently the compiler specificaly passes only the `href` when it sees a namespaced attribute belonging to a foreign element (svg,...), so there may be a reason behind it that I don't understand yet

Reply:
I'm not sure why too, maybe @natemoo-re knows checking the git blame 👀

Reply:
Yep, I'm looking into this now!

Reply:
OK it was just a bug in the namespace logic. Should be fixed by my commit with no change to `astro` itself!